### PR TITLE
Format the world

### LINKED ISSUE #45: Automatic code formatting
From the quick testing I did, none of the tools seem to work well with the current code style.

### PR DESCRIPTION
**Pull request by GnomedDev**
This PR adds CI for checking `clang-format` has been run, then formats the entire project. I also did the liberty of cleaning up some sections that look worse and/or create errors with formatting changes.

Closes #45

### CODE REVIEW

**Comment by vabold** [comment]
> due to it's length
Should be "its". Gotta throw the whole PR away unfortunately.

**Comment by GnomedDev** [comment]
I have a personal vendetta against the English language for that one, same with sentral.

**Comment by GnomedDev** [comment]
Ah, knew there was a tracking issue for this, closes #45

File: payload/sp/storage/Sdi.c [hunk 1]
```diff
@@ -1,7 +1,8 @@
 // Resources:
 // - https://wiibrew.org/wiki//dev/sdio
 // - https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Core/IOS/SDIO/SDIOSlot0.cpp
-// - https://www.sdcard.org/cms/wp-content/themes/sdcard-org/dl.php?f=Part1_Physical_Layer_Simplified_Specification_Ver8.00.pdf
+// -
+// https://www.sdcard.org/cms/wp-content/themes/sdcard-org/dl.php?f=Part1_Physical_Layer_Simplified_Specification_Ver8.00.pdf
```
Comment (stblr) [inline payload/sp/storage/Sdi.c:5]: Not sure what to do about these links.

File: payload/sp/storage/UsbStorage.c [hunk 2]
```diff
@@ -1,7 +1,9 @@
 // Resources:
 // - https://www.usb.org/sites/default/files/usbmassbulk_10.pdf
-// - https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
-// - https://www.downtowndougbrown.com/2018/12/usb-mass-storage-with-embedded-devices-tips-and-quirks/
+// -
+// https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+// -
+// https://www.downtowndougbrown.com/2018/12/usb-mass-storage-with-embedded-devices-tips-and-quirks/
```
Comment (stblr) [inline payload/sp/storage/UsbStorage.c:6]: Same here.

File: payload/sp/storage/Sdi.c (hunk 1 above)
Comment (GnomedDev) [reply payload/sp/storage/Sdi.c:5]: Do you want me to chuck a `clang-format off`? Remove the dashes?

Comment (stblr) [reply payload/sp/storage/Sdi.c:5]: Maybe the former like in `SIKeyboard.c`.

**Review by stblr — APPROVED** [review]

**Pull request merged**